btclib/src/types/block.rs

```diff
@@ -0,0 +1,211 @@
+    header: BlockHeader,
+    /// List of transactions included in this block
+    transactions: Vec<Transaction>,
+}
+
+impl BlockHeader {
+    pub fn new(version: u32, prev_block_hash: [u8; 32], merkle_root: [u8; 32], target: u32) -> Self {
+        Self {
+            version,
+            timestamp: SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .unwrap()
+                .as_secs(),
+            prev_block_hash,
+            merkle_root,
+            target,
+            nonce: 0,
+        }
+    }
+
+    /// Increment the nonce value - used during mining
+    pub fn increment_nonce(&mut self) {
+        self.nonce = self.nonce.wrapping_add(1);
+    }
+
+    /// Calculate the hash of this block header
+    pub fn hash(&self) -> [u8; 32] {
+        let serialized = bincode::serialize(&self).unwrap();
+        let mut hasher = Sha256::new();
+        hasher.update(&serialized);
+        let result = hasher.finalize();
+        let mut hash = [0u8; 32];
+        hash.copy_from_slice(&result);
+        hash
+    }
+}
+
+impl Block {
+    pub fn new(
+        version: u32,
+        prev_block_hash: [u8; 32],
+        transactions: Vec<Transaction>,
+        target: u32,
+    ) -> Self {
+        // Calculate merkle root from transactions
+        let merkle_root = Self::calculate_merkle_root(&transactions);
+        
+        Self {
+            header: BlockHeader::new(version, prev_block_hash, merkle_root, target),
+            transactions,
+        }
+    }
+
+    /// Calculate the merkle root of the transactions
+    fn calculate_merkle_root(transactions: &[Transaction]) -> [u8; 32] {
+        // Serialize transactions for merkle tree
+        let tx_bytes: Vec<Vec<u8>> = transactions
+            .iter()
+            .map(|tx| bincode::serialize(&tx).unwrap())
+            .collect();
+
+        // Create merkle tree and get root hash
+        let tree = MerkleTree::new(&tx_bytes);
+        tree.root_hash().unwrap_or([0u8; 32])
+    }
+
+    /// Get reference to transactions
+    pub fn transactions(&self) -> &[Transaction] {
+        &self.transactions
+    }
+
+    /// Get block height (to be implemented with chain context)
+    pub fn height(&self) -> u64 {
+        // TODO: Implement proper height tracking
+        0
+    }
+
+    /// Validate basic block properties
+    pub fn validate(&self) -> bool {
+        // Verify proof of work
+        let hash = self.header.hash();
+        let target = self.header.target;
+        
+        // Convert first 4 bytes of hash to u32 for difficulty comparison
+        let hash_value = u32::from_be_bytes([hash[0], hash[1], hash[2], hash[3]]);
+        
+        // Check if hash meets difficulty target
+        if hash_value > target {
+            return false;
+        }
+
+        // Verify merkle root matches transactions
+        let calculated_root = Self::calculate_merkle_root(&self.transactions);
+        if calculated_root != self.header.merkle_root {
```

Comment:
Think these could be resumed as returning just equality:

```suggestion
        calculated_root == self.header.merkle_root
```

And not needing that `true` return at the end.
But's it's subjective.